db/migrate/20230920055719_add_state_column_to_tipline_message.rb

```diff
@@ -0,0 +1,10 @@
+class AddStateColumnToTiplineMessage < ActiveRecord::Migration[6.1]
+  def change
+    add_column :tipline_messages, :state, :string
+    remove_index :tipline_messages, name: "index_tipline_messages_on_external_id"
+    add_index :tipline_messages, :external_id
```

Comment:
Why you are removing and re-adding the same index? Is it because the previous one was a unique index and this one is not? Well, if so, yes, we may have the same `external_id` for two `TiplineMessage`'s, but one with `sent` state and another with `delivered` state, for example... so I suggest we replace this `add_index` by a unique index for `external_id, state`. This may need to happen after the `update_all` below.

Reply:
For now I keep the regular index for `external_id` column only as I found many tests need to fix due to unique index for `external_id, state` but locally I see the `external_id, state` are unique

Reply:
Hopefully there is something in a helper that can reduce the number of changes needed in the tests (probably many are using a hard-coded ID `123456` and we should replace by `random_string`), but it's important to add the unique index because I've seen WhatsApp sending the same notification for the same event so we need to guarantee that these are unique since they are used for statistics.